net-analyzer/openvas Version bump to 9.0.0 + New Package + Maintainership

hsntgm opened the pull request:
Hi Juippis,
This is my third attempt. Fixed all major things.

> Oh boy, this is a huge PR. There's a lot to be done before this one is mergeable. I'd scrap this and start again.
> 
> 1: I checked all of the projects you pushed, they have releases in their github pages. You can use the release .tar.gz instead of pulling straight from git using git-r3.eclass. There even was one ebuild without git or SRC_URI at all, did you test these all work the way you intend to?
> 2: Learn to use repoman, and especially commit with it to satisfy many required things in GLEP-66 (linked above).
> 3: Missing sign-off, check the link above for GLEP-76.
> 4: When committing new packages, you can commit one new directory with new commit. There's no need to make a new commit for every single file. So instead of having 65 commits in here, this can be squashed to I'm guessing 11?
> 5: (Personal preference:) You should work in branches rather than in your git master. If this is your first time working with git, I'd perhaps start with something easier until the flow gets familiar. There are a lot of guides offered by Gentoo wiki how to work in git branches and do proper pull requests. Also, you can maybe test setting this all up in an overlay?
> 
> There are few major notes, I'll get into minor ones when the bigger are dealt with. As said, lot to be done, but it can be a nice learning experience and overall the quality of ebuilds seem to look good, so don't give up!

juippis commented on net-analyzer/openvas-scanner/openvas-scanner-5.1.3.ebuild:
```diff
@@ -0,0 +1,85 @@
+	cmake-utils_src_prepare
+	sed -i "s*/doc/openvas-scanner/*/doc/openvas-scanner-${PV}/*g" "$S"/CMakeLists.txt || die
+	if use extras; then
+		doxygen -u "$S"/doc/Doxyfile_full.in
```
Append `|| die` and add doxygen to BDEPEND, you know the thrill.

juippis commented on net-analyzer/openvas-scanner/openvas-scanner-5.1.3.ebuild:
```diff
@@ -0,0 +1,85 @@
+
+src_install() {
+	cmake-utils_src_install
+	einstalldocs
```
Should be called already.

juippis commented on net-analyzer/openvas/metadata.xml:
```diff
@@ -1,8 +1,21 @@
+	</maintainer>
+	<use>
+		<flag name="extras">Fonts,pdf results and docs support</flag>
+		<flag name="cli">Command Line Interface for OpenVAS Scanner throught <pkg>net-analyzer/gvm-tools</pkg></flag>
```
Typo.

juippis commented on net-analyzer/openvas/metadata.xml:
```diff
@@ -1,8 +1,21 @@
+		<name>Proxy Maintainers</name>
+	</maintainer>
+	<use>
+		<flag name="extras">Fonts,pdf results and docs support</flag>
```
`Fonts, pdf`

juippis commented on net-analyzer/openvas/openvas-9.0.0.ebuild:
```diff
@@ -0,0 +1,46 @@
+KEYWORDS="~amd64 ~x86"
+IUSE="+extras +cli +gsa +ospd ldap radius"
+
+DEPEND="
```
This ebuild doesn't seem to compile anything, so this DEPEND is definitely wrong. Also why should this meta package pull all that stuff in extras? The ebuilds themself should pull all those.

hsntgm replied:
Ok.This is the biggest one i think.

This workflow effects all ebuilds and it is a bit tricky to handle this.
Let me explain and please correct me if i do something wrong.

This is main OpenVAS ebuild (openvas-9.0.0).So i suppose ( and old maintainer too:) ) if someone needs this package will emerge this one first.This one calls other OpenVAS components for building (ordered | libraries-->scanner-->manager-->security-agent-->ospd-->gvm-tools).

So DEPEND is setted to OpenVAS components.All openvas components have seperate ebuilds because of separate dependencies but same for doc,html support ('extras' -use flag)

OpenVAS Components;

> OpenVAS-Libraries
> OpenVAS-Scanner
> OpneVAS-Manager
> Greenbone-Security-Agent 
> Gvm-Tools
> ospd

OpenVAS-Libraries is the key ebuild which other components depend this one build time.So if someone try to emerge OpenVAS components directly e.g. manager, scanner, security agent, ebuilds will call OpenVAS-Libraries.So all direct emerge attempts will point to OpenVAS-Libraries for successfull build.

Here is the GAP:
----------------------
If someone emerge OpenVAS-Libraries directly instead of Openvas main, whats then? Main OpenVAS ebuild is meaningless here.

Then he/she has to continue emerging all components manually one by one and ordered
-->scanner-->manager-->security-agent e.g. to complete setup which is not logical.

Fixing it? How to?
If i make openvas-libraries-9.0.3 depended to net-analyzer/openvas-9.0.0 for pointing all direct component emerges to main package.This is a bit funny loop while removing package:)

openvas-9.0.0.ebuild
```
DEPEND="
        >=net-analyzer/openvas-libraries-9.0.3
"
```

openvas-libraries-9.0.3.ebuild
```
DEPEND="
        >=net-analyzer/openvas-9.0.0
"
```

juippis replied:
These circular dependencies can be fixed with PDEPEND.

Check the devmanual and examples from Gentoo tree how it works. If you can't find a solution using that, let me know and let's think of something else.

juippis commented on net-analyzer/ospd/ospd-1.3.2.ebuild:
```diff
@@ -0,0 +1,47 @@
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_6 )
```
Does it really only support python3.6? Same comments as in the previous python-related package.

hsntgm replied:
https://github.com/greenbone/ospd/blob/ospd-1.3/INSTALL
Prerequisites:
* python 3

juippis replied:
But remember that Gentoo provides python3.4, python3.5, python3.6 and python3.7 currently. python3.6 is the default and always a safe bat, but IF your program works with other versions, you should enable them for the user. distutils-r1 has DISTUTILS_SINGLE_IMPL variable to use setuptools but install for one implementation only.

juippis commented on net-analyzer/ospd/ospd-1.3.2.ebuild:
```diff
@@ -0,0 +1,47 @@
+EAPI=7
+
+PYTHON_COMPAT=( python3_6 )
+inherit eutils distutils-r1
```
eutils not used :)

juippis commented on net-analyzer/ospd/ospd-1.3.2.ebuild:
```diff
@@ -0,0 +1,47 @@
+IUSE="extras"
+
+DEPEND="
+	dev-python/setuptools
```
setuptools in (B)DEPEND is already covered by distutils-r1.eclass

juippis commented on net-analyzer/ospd/ospd-1.3.2.ebuild:
```diff
@@ -0,0 +1,47 @@
+
+DEPEND="
+	dev-python/setuptools
+	>=dev-python/paramiko-2.4.2
```
Missing python_deps.

juippis commented on net-analyzer/ospd/ospd-1.3.2.ebuild:
```diff
@@ -0,0 +1,47 @@
+DEPEND="
+	dev-python/setuptools
+	>=dev-python/paramiko-2.4.2
+	dev-python/defusedxml
```
Missing python_deps.

juippis commented on net-analyzer/ospd/ospd-1.3.2.ebuild:
```diff
@@ -0,0 +1,47 @@
+	dev-python/setuptools
+	>=dev-python/paramiko-2.4.2
+	dev-python/defusedxml
+	dev-python/lxml
```
Missing python_deps.

juippis commented on net-analyzer/ospd/ospd-1.3.2.ebuild:
```diff
@@ -0,0 +1,47 @@
+
+BDEPEND="virtual/pkgconfig"
+
+DOCS=( CHANGES COPYING README.md )
```
Should be part of default einstalldocs.

juippis commented on net-analyzer/ospd/ospd-1.3.2.ebuild:
```diff
@@ -0,0 +1,47 @@
+}
+
+python_install_all() {
+	einstalldocs
```
Should be called by distutils-r1_src_install on its own.

juippis commented on net-analyzer/ospd/ospd-1.3.2.ebuild:
```diff
@@ -0,0 +1,47 @@
+
+python_compile() {
+	if use extras; then
+		/bin/bash "${S}"/doc/generate
```
Will this work with prefix systems? I think you can safely just call bash, OR just execute the generate script itself. Also append with `|| die`.

juippis replied:
This still needs to be addressed.

hsntgm replied:
done now.

juippis commented:
Heya. I see most of my comments are repetition, you do same mistakes in every ebuild. `einstalldocs` is part of EAPI since 6, so you dont have to inherit eutils to use that. Then, doxygen belongs to BDEPEND etc what I mentioned. Nothing really bad stands out there, so forgive me about the spam. Looks good, just some fine grinding to do!

hsntgm commented:
Thank you for your all reviews juippis.Your all comments is so clear.I will be looking for all minor things.

These are my common mistakes, i will fix them easily;

1)What i see that i use eutils wrong. I see `einstalldocs` is part of EAPI since 6.I will remove unnecessary eclass.
2)Sorting package lists and eclasses alphabetically.
3)Unnecessary DOCS variable.Without it 'einstalldocs' will do job.
4)missing || die in some ebuilds.
5)Typo issues.

For other specific issues and questions i will be commenting one by one.

At this point; I want to ask about pull request update.Maybe there is an other procedure for gentoo i missed. I will follow this workflow to update pull request ;

update my local branch, 
commit new changes, 
push to github branch.

This will automatically update pull request too.
Do i need to add "Correcting for PR comments" or something equal to my commit's messages?

Thanks.

juippis commented:
> Do i need to add "Correcting for PR comments" or something equal to my commit's messages?
> 

Ideally, you would rebase and squash your commits back to the original commit. So you do fixes, but the commit count wont increase. It might be a bit difficult if you've never done it before, but it's basically just reordering commits in the editor when rebasing. Then use 'fixup' or 'squash', to include them in the original commits.

hsntgm commented on net-analyzer/openvas-libraries/openvas-libraries-9.0.3.ebuild:
```diff
@@ -0,0 +1,90 @@
+	dev-perl/UUID
+	sys-libs/zlib
+	>=dev-libs/glib-2.32
+	>=dev-libs/hiredis-0.10.1
```
Lifted off unnecessary package requirements.
	dev-libs/hiredis
	dev-libs/libksba
	net-libs/gnutls
	net-libs/libssh

hsntgm commented on net-analyzer/openvas-libraries/openvas-libraries-9.0.3.ebuild:
```diff
@@ -0,0 +1,90 @@
+	sys-libs/zlib
+	>=dev-libs/glib-2.32
+	>=dev-libs/hiredis-0.10.1
+	>=dev-libs/libgcrypt-1.6
```
subslot binding 
`>=dev-libs/libgcrypt-1.6:=`

hsntgm commented on net-analyzer/openvas-libraries/openvas-libraries-9.0.3.ebuild:
```diff
@@ -0,0 +1,90 @@
+	>=dev-libs/libgcrypt-1.6
+	>=dev-libs/libksba-1.0.7
+	net-analyzer/net-snmp
+	>=net-libs/gnutls-3.2.15[tools]
```
`net-libs/gnutls:=[tools]`

hsntgm commented on net-analyzer/openvas-libraries/openvas-libraries-9.0.3.ebuild:
```diff
@@ -0,0 +1,90 @@
+	net-analyzer/net-snmp
+	>=net-libs/gnutls-3.2.15[tools]
+	net-libs/libpcap
+	>=net-libs/libssh-0.5.0
```
`net-libs/libssh:=`

hsntgm commented on net-analyzer/greenbone-security-assistant/greenbone-security-assistant-7.0.3.ebuild:
```diff
@@ -0,0 +1,83 @@
+
+RDEPEND="
+	dev-libs/libxslt
+	dev-libs/libxml2
```
```
 dev-libs/libxml2:2
 dev-libs/libxslt
```

hsntgm commented on net-analyzer/greenbone-security-assistant/greenbone-security-assistant-7.0.3.ebuild:
```diff
@@ -0,0 +1,83 @@
+	net-libs/libmicrohttpd[messages]"
+
+DEPEND="${RDEPEND}
+	virtual/pkgconfig"
```
```
DEPEND="
        dev-libs/libxml2:2
        dev-libs/libxslt
        >=net-analyzer/openvas-libraries-9.0.3
        net-libs/libmicrohttpd[messages]"

RDEPEND="${DEPEND}"

BDEPEND="virtual/pkgconfig"
```

hsntgm commented on net-analyzer/openvas-manager/openvas-manager-7.0.3.ebuild:
```diff
@@ -0,0 +1,83 @@
+IUSE="extras"
+
+DEPEND="
+	>=dev-db/sqlite-3
```
package name already wrong.slotted.
`dev-db/sqlite:3`

hsntgm commented on net-analyzer/openvas-manager/openvas-manager-7.0.3.ebuild:
```diff
@@ -0,0 +1,83 @@
+
+DEPEND="
+	>=dev-db/sqlite-3
+	extras? ( media-libs/gd
```
You are right 2/3
`media-libs/gd:2=`

hsntgm commented on net-analyzer/openvas-manager/openvas-manager-7.0.3.ebuild:
```diff
@@ -0,0 +1,83 @@
+		"-DLOCALSTATEDIR=${EPREFIX}/var"
+		"-DSYSCONFDIR=${EPREFIX}/etc"
+	)
+	append-cflags -Wno-nonnull
```
simply we can call this "Disable-the-non-null-compare-warning-error.patch"
There is no any bug report on bugzilla.gentoo or upstream about that.
Without this flag build works but gentoo warns me something like that 'encountered runtime issues while compiling'. I looked code deeply and it seems this issue false positive and no harm anything.So i removed gentoo warning with this flag.

juippis replied:
I don't think it needs a patch. Just add something like 
`# Fix runtime QA error for 7.0.3
append-cflags -Wno-nonnull`

PS It should be reported, and fixed, upstream if possible.

hsntgm commented on net-analyzer/gvm-tools/gvm-tools-1.4.1.ebuild:
```diff
@@ -0,0 +1,48 @@
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_6 )
```
https://github.com/greenbone/gvm-tools/blob/master/README.md#requirements
Python 3.5 and later is supported.

hsntgm commented on net-analyzer/gvm-tools/gvm-tools-1.4.1.ebuild:
```diff
@@ -0,0 +1,48 @@
+	>=dev-python/lxml-4.2.5
+	>=dev-python/paramiko-2.4.2
+	>=dev-python/defusedxml-0.5.0
+	dev-python/pythondialog:0
```
There is python-2 slot.

3.4.0-r200 : python-2 
3.4.0 : 0 
3.3.0-r200 : python-2
3.3.0 : 0
3.2.2 : 0

juippis replied:
Oh, I wasn't aware. Cheers :)

juippis commented:
So, not quite. You needed to rebase and fix it with the rebase editor. So, what I guess you need to do, is:

`git rebase -i HEAD~14`

Then you will see an editor. It should look something like this currently: 

```
keep net-analyzer/openvas: version bump to 9.0.0.				a0bde23
keep net-analyzer/openvas-libraries: version bump to 9.0.3.			61e9629
keep net-analyzer/openvas-scanner: version bump to 5.1.3.			596d0b1
keep net-analyzer/greenbone-security-assistant: version bump to 7.0.3.		775b193
keep net-analyzer/openvas-manager: version bump to 7.0.3. 			26a5f6d
keep net-analyzer/ospd: version bump to 1.3.2. 					796c24d
keep net-analyzer/gvm-tools: New package.					5d607b9
keep fixup! net-analyzer/openvas: version bump to 9.0.0. 			1edeeda
keep fixup! net-analyzer/openvas-libraries: version bump to 9.0.3. 		21506da
keep fixup! net-analyzer/openvas-scanner: version bump to 5.1.3.		1c49350
keep fixup! net-analyzer/greenbone-security-assistant: version bump to 7.0.3.	eb1fe73
keep fixup! net-analyzer/openvas-manager: version bump to 7.0.3.		d11e760
keep fixup! net-analyzer/ospd: version bump to 1.3.2. 				5a95cd2
keep fixup! net-analyzer/gvm-tools: New package.				43fcede 
```

And what you wanna do, is re-order the commits so a fix comes after initial commit, and you want to "fixup" your fixed commit into the original one. So within your editor, you need to sort the commits, and update "keep" with "fixup". So in the end, it's going to look something like this:

```
keep net-analyzer/openvas: version bump to 9.0.0.				a0bde23
fixup fixup! net-analyzer/openvas: version bump to 9.0.0. 			1edeeda
keep net-analyzer/openvas-libraries: version bump to 9.0.3.			61e9629
fixup fixup! net-analyzer/openvas-libraries: version bump to 9.0.3. 		21506da
keep net-analyzer/openvas-scanner: version bump to 5.1.3.			596d0b1
fixup fixup! net-analyzer/openvas-scanner: version bump to 5.1.3.		1c49350
keep net-analyzer/greenbone-security-assistant: version bump to 7.0.3.		775b193
fixup fixup! net-analyzer/greenbone-security-assistant: version bump to 7.0.3.	eb1fe73
keep net-analyzer/openvas-manager: version bump to 7.0.3. 			26a5f6d
fixup fixup! net-analyzer/openvas-manager: version bump to 7.0.3.		d11e760
keep net-analyzer/ospd: version bump to 1.3.2. 					796c24d
fixup fixup! net-analyzer/ospd: version bump to 1.3.2. 				5a95cd2
keep net-analyzer/gvm-tools: New package.					5d607b9
fixup fixup! net-analyzer/gvm-tools: New package.				43fcede
```

I'm not gonna say it's going to look exactly like that, but you get the idea! Be careful and be precise. After you're done, save and exit editor, your commits should then be rebased. Do a force push to update this PR. I will then take another look. If you fail somewhere, you can do a hard git reset locally and copypaste the complete files from this Github web UI, and do fresh new commits with force push.

hsntgm commented:
Thank you juippis.You forwarding me great.

Ok i got it.I thought that you will review fixups seperately first.
So i didn't rebased and squashed them first time.

After your reviews i deeply revised all ebuilds and found/fixed some other issues too.Sorry for new changes we haven't been discussed yet.I am ready to discuss them.

I created overlay for clean view and maybe someone wanna make quick test.
https://github.com/hsntgm/openvas-9-gentoo-overlay

*For flag-o-matic cflag issue i will inform upstream as soon as possible about issue.
I informed them about insecure rpath issue already and they said that "we need bigger patch than yours" :)
https://github.com/greenbone/gvm-libs/pulls

*All my patches taken from upstream expect rpath,underlinking for openvas-libraries.If you need them i can share upstream links.
*The scripts in openvas-scanner is written by me and fully clean.
*I tested PDEPEND and it seems it do the job well.But it need your review.

Thank you.

hsntgm commented:
for openvas-manager ebuild (cflag append -Wno-nonnull) -- >>  build warning reported to upstream with temprorary fix.
https://github.com/greenbone/gvmd/issues/395

hsntgm commented:
ping

juippis commented:
Sorry, havent been at my computer ~at all lately. Out of curiosity, why you set `CMAKE_MAKEFILE_GENERATOR="emake"`, isn't that the default value? Or is it advised to set now that ninja is starting to take over? 

I saw some ~minor things there, but it doesn't stop me from testing, so I'll do testing rounds tomorrow (in ~24 h) and report back, and comment on the minor things as well.

hsntgm commented:
Yes,default ninja is starting to take over and this is not package's default build system.I prefered emake to stay in safe sea.

juippis commented on net-analyzer/greenbone-security-assistant/greenbone-security-assistant-7.0.3.ebuild:
```diff
@@ -0,0 +1,93 @@
+	${DEPEND}"
+
+BDEPEND="
+	app-doc/doxygen[dot]
```
You probably want to make this conditional with extras? USE flag, as you've done with all other ebuilds.

hsntgm replied:
You are right.Fixed.

juippis commented on net-analyzer/greenbone-security-assistant/greenbone-security-assistant-7.0.3.ebuild:
```diff
@@ -0,0 +1,93 @@
+	dev-libs/libxslt
+	extras? (
+		dev-python/polib
+		sys-devel/gettext
```
This is also BDEPEND.

juippis commented on net-analyzer/openvas-scanner/openvas-scanner-5.1.3.ebuild:
```diff
@@ -0,0 +1,97 @@
+	!net-analyzer/openvas-tools"
+
+BDEPEND="
+	app-doc/doxygen[dot]
```
Depend conditionally with extras? UsE flag.

hsntgm replied:
You are right.Fixed.

juippis commented:
You probably should remove the PDEPEND part from any other ebuild except openvas-libraries, since the other ebuilds depend on the libraries already that PDEPENDs on openvas.

juippis commented:
```
Error: translate: Error with producer 'SQL::Translator::Producer::HTML': Can't locate CGI/Pretty.pm in @INC (you may need to install the CGI::Pretty module) (@INC contains: /etc/perl /usr/local/lib64/perl5/5.28.0/x86_64-linux /usr/local/lib64/perl5/5.28.0 /usr/lib64/perl5/vendor_perl/5.28.0/x86_64-linux /usr/lib64/perl5/vendor_perl/5.28.0 /usr/local/lib64/perl5 /usr/lib64/perl5/vendor_perl/5.26.2 /usr/lib64/perl5/vendor_perl/5.26.1 /usr/lib64/perl5/vendor_perl/5.24.0 /usr/lib64/perl5/vendor_perl/5.22.0 /usr/lib64/perl5/vendor_perl/5.20.2 /usr/lib64/perl5/vendor_perl /usr/lib64/perl5/5.28.0/x86_64-linux /usr/lib64/perl5/5.28.0) at /usr/lib64/perl5/vendor_perl/5.28.0/SQL/Translator/Producer/HTML.pm line 37.
make[3]: *** [doc/CMakeFiles/doc.dir/build.make:76: doc/.built-html_db] Error 2
make[3]: *** Waiting for unfinished jobs....
Image written to 'db.png'.  Done.
make[3]: Leaving directory '/tmp/portage/net-analyzer/openvas-manager-7.0.3/work/gvmd-7.0.3_build'
make[2]: *** [CMakeFiles/Makefile2:539: doc/CMakeFiles/doc.dir/all] Error 2
make[2]: Leaving directory '/tmp/portage/net-analyzer/openvas-manager-7.0.3/work/gvmd-7.0.3_build'
make[1]: *** [CMakeFiles/Makefile2:546: doc/CMakeFiles/doc.dir/rule] Error 2
make[1]: Leaving directory '/tmp/portage/net-analyzer/openvas-manager-7.0.3/work/gvmd-7.0.3_build'
make: *** [Makefile:342: doc] Error 2
```

Seems like "extras" for openvs-manager will require `dev-perl/CGI` in DEPEND.

juippis commented:
There was a QA notice from either openvas or openvas-libraries about unused CMAKE variable, CMAKE_BUILD_RADIUS I believe. Could you investigate that? 

gvm-tools gives:
```
 * The ebuild is installing to one or more unexpected paths:
 * 
 *   /usr/VERSION
 * 
 * Please fix the ebuild to use correct FHS/Gentoo policy paths.
```
```
 * Contents of net-analyzer/gvm-tools-1.4.1:
/usr
/usr/VERSION
/usr/bin
/usr/bin/gvm-cli -> ../lib/python-exec/python-exec2
/usr/bin/gvm-dialog -> ../lib/python-exec/python-exec2
/usr/bin/gvm-pyshell -> ../lib/python-exec/python-exec2
/usr/lib
/usr/lib/python-exec
/usr/lib/python-exec/python3.5
/usr/lib/python-exec/python3.5/gvm-cli
/usr/lib/python-exec/python3.5/gvm-dialog
/usr/lib/python-exec/python3.5/gvm-pyshell
/usr/lib/python-exec/python3.6
/usr/lib/python-exec/python3.6/gvm-cli
/usr/lib/python-exec/python3.6/gvm-dialog
/usr/lib/python-exec/python3.6/gvm-pyshell
/usr/lib64
/usr/lib64/python3.5
/usr/lib64/python3.5/site-packages
/usr/lib64/python3.5/site-packages/gmp
/usr/lib64/python3.5/site-packages/gmp/__init__.py
/usr/lib64/python3.5/site-packages/gmp/__pycache__
/usr/lib64/python3.5/site-packages/gmp/__pycache__/__init__.cpython-35.opt-1.pyc
/usr/lib64/python3.5/site-packages/gmp/__pycache__/__init__.cpython-35.opt-2.pyc
/usr/lib64/python3.5/site-packages/gmp/__pycache__/__init__.cpython-35.pyc
/usr/lib64/python3.5/site-packages/gmp/__pycache__/gmp.cpython-35.opt-1.pyc
/usr/lib64/python3.5/site-packages/gmp/__pycache__/gmp.cpython-35.opt-2.pyc
/usr/lib64/python3.5/site-packages/gmp/__pycache__/gmp.cpython-35.pyc
/usr/lib64/python3.5/site-packages/gmp/__pycache__/gvm_connection.cpython-35.opt-1.pyc
/usr/lib64/python3.5/site-packages/gmp/__pycache__/gvm_connection.cpython-35.opt-2.pyc
/usr/lib64/python3.5/site-packages/gmp/__pycache__/gvm_connection.cpython-35.pyc
/usr/lib64/python3.5/site-packages/gmp/__pycache__/helper.cpython-35.opt-1.pyc
/usr/lib64/python3.5/site-packages/gmp/__pycache__/helper.cpython-35.opt-2.pyc
/usr/lib64/python3.5/site-packages/gmp/__pycache__/helper.cpython-35.pyc
/usr/lib64/python3.5/site-packages/gmp/clients
/usr/lib64/python3.5/site-packages/gmp/clients/__init__.py
/usr/lib64/python3.5/site-packages/gmp/clients/__pycache__
/usr/lib64/python3.5/site-packages/gmp/clients/__pycache__/__init__.cpython-35.opt-1.pyc
/usr/lib64/python3.5/site-packages/gmp/clients/__pycache__/__init__.cpython-35.opt-2.pyc
/usr/lib64/python3.5/site-packages/gmp/clients/__pycache__/__init__.cpython-35.pyc
/usr/lib64/python3.5/site-packages/gmp/clients/__pycache__/gvm_cli.cpython-35.opt-1.pyc
/usr/lib64/python3.5/site-packages/gmp/clients/__pycache__/gvm_cli.cpython-35.opt-2.pyc
/usr/lib64/python3.5/site-packages/gmp/clients/__pycache__/gvm_cli.cpython-35.pyc
/usr/lib64/python3.5/site-packages/gmp/clients/__pycache__/gvm_dialog.cpython-35.opt-1.pyc
/usr/lib64/python3.5/site-packages/gmp/clients/__pycache__/gvm_dialog.cpython-35.opt-2.pyc
/usr/lib64/python3.5/site-packages/gmp/clients/__pycache__/gvm_dialog.cpython-35.pyc
/usr/lib64/python3.5/site-packages/gmp/clients/__pycache__/gvm_pyshell.cpython-35.opt-1.pyc
/usr/lib64/python3.5/site-packages/gmp/clients/__pycache__/gvm_pyshell.cpython-35.opt-2.pyc
/usr/lib64/python3.5/site-packages/gmp/clients/__pycache__/gvm_pyshell.cpython-35.pyc
/usr/lib64/python3.5/site-packages/gmp/clients/gvm_cli.py
/usr/lib64/python3.5/site-packages/gmp/clients/gvm_dialog.py
/usr/lib64/python3.5/site-packages/gmp/clients/gvm_pyshell.py
/usr/lib64/python3.5/site-packages/gmp/gmp.py
/usr/lib64/python3.5/site-packages/gmp/gvm_connection.py
/usr/lib64/python3.5/site-packages/gmp/helper.py
/usr/lib64/python3.5/site-packages/gvm_tools-1.4.1-py3.5.egg-info
/usr/lib64/python3.5/site-packages/gvm_tools-1.4.1-py3.5.egg-info/PKG-INFO
/usr/lib64/python3.5/site-packages/gvm_tools-1.4.1-py3.5.egg-info/SOURCES.txt
/usr/lib64/python3.5/site-packages/gvm_tools-1.4.1-py3.5.egg-info/dependency_links.txt
/usr/lib64/python3.5/site-packages/gvm_tools-1.4.1-py3.5.egg-info/entry_points.txt
/usr/lib64/python3.5/site-packages/gvm_tools-1.4.1-py3.5.egg-info/requires.txt
/usr/lib64/python3.5/site-packages/gvm_tools-1.4.1-py3.5.egg-info/top_level.txt
/usr/lib64/python3.6
/usr/lib64/python3.6/site-packages
/usr/lib64/python3.6/site-packages/gmp
/usr/lib64/python3.6/site-packages/gmp/__init__.py
/usr/lib64/python3.6/site-packages/gmp/__pycache__
/usr/lib64/python3.6/site-packages/gmp/__pycache__/__init__.cpython-36.opt-1.pyc
/usr/lib64/python3.6/site-packages/gmp/__pycache__/__init__.cpython-36.opt-2.pyc
/usr/lib64/python3.6/site-packages/gmp/__pycache__/__init__.cpython-36.pyc
/usr/lib64/python3.6/site-packages/gmp/__pycache__/gmp.cpython-36.opt-1.pyc
/usr/lib64/python3.6/site-packages/gmp/__pycache__/gmp.cpython-36.opt-2.pyc
/usr/lib64/python3.6/site-packages/gmp/__pycache__/gmp.cpython-36.pyc
/usr/lib64/python3.6/site-packages/gmp/__pycache__/gvm_connection.cpython-36.opt-1.pyc
/usr/lib64/python3.6/site-packages/gmp/__pycache__/gvm_connection.cpython-36.opt-2.pyc
/usr/lib64/python3.6/site-packages/gmp/__pycache__/gvm_connection.cpython-36.pyc
/usr/lib64/python3.6/site-packages/gmp/__pycache__/helper.cpython-36.opt-1.pyc
/usr/lib64/python3.6/site-packages/gmp/__pycache__/helper.cpython-36.opt-2.pyc
/usr/lib64/python3.6/site-packages/gmp/__pycache__/helper.cpython-36.pyc
/usr/lib64/python3.6/site-packages/gmp/clients
/usr/lib64/python3.6/site-packages/gmp/clients/__init__.py
/usr/lib64/python3.6/site-packages/gmp/clients/__pycache__
/usr/lib64/python3.6/site-packages/gmp/clients/__pycache__/__init__.cpython-36.opt-1.pyc
/usr/lib64/python3.6/site-packages/gmp/clients/__pycache__/__init__.cpython-36.opt-2.pyc
/usr/lib64/python3.6/site-packages/gmp/clients/__pycache__/__init__.cpython-36.pyc
/usr/lib64/python3.6/site-packages/gmp/clients/__pycache__/gvm_cli.cpython-36.opt-1.pyc
/usr/lib64/python3.6/site-packages/gmp/clients/__pycache__/gvm_cli.cpython-36.opt-2.pyc
/usr/lib64/python3.6/site-packages/gmp/clients/__pycache__/gvm_cli.cpython-36.pyc
/usr/lib64/python3.6/site-packages/gmp/clients/__pycache__/gvm_dialog.cpython-36.opt-1.pyc
/usr/lib64/python3.6/site-packages/gmp/clients/__pycache__/gvm_dialog.cpython-36.opt-2.pyc
/usr/lib64/python3.6/site-packages/gmp/clients/__pycache__/gvm_dialog.cpython-36.pyc
/usr/lib64/python3.6/site-packages/gmp/clients/__pycache__/gvm_pyshell.cpython-36.opt-1.pyc
/usr/lib64/python3.6/site-packages/gmp/clients/__pycache__/gvm_pyshell.cpython-36.opt-2.pyc
/usr/lib64/python3.6/site-packages/gmp/clients/__pycache__/gvm_pyshell.cpython-36.pyc
/usr/lib64/python3.6/site-packages/gmp/clients/gvm_cli.py
/usr/lib64/python3.6/site-packages/gmp/clients/gvm_dialog.py
/usr/lib64/python3.6/site-packages/gmp/clients/gvm_pyshell.py
/usr/lib64/python3.6/site-packages/gmp/gmp.py
/usr/lib64/python3.6/site-packages/gmp/gvm_connection.py
/usr/lib64/python3.6/site-packages/gmp/helper.py
/usr/lib64/python3.6/site-packages/gvm_tools-1.4.1-py3.6.egg-info
/usr/lib64/python3.6/site-packages/gvm_tools-1.4.1-py3.6.egg-info/PKG-INFO
/usr/lib64/python3.6/site-packages/gvm_tools-1.4.1-py3.6.egg-info/SOURCES.txt
/usr/lib64/python3.6/site-packages/gvm_tools-1.4.1-py3.6.egg-info/dependency_links.txt
/usr/lib64/python3.6/site-packages/gvm_tools-1.4.1-py3.6.egg-info/entry_points.txt
/usr/lib64/python3.6/site-packages/gvm_tools-1.4.1-py3.6.egg-info/requires.txt
/usr/lib64/python3.6/site-packages/gvm_tools-1.4.1-py3.6.egg-info/top_level.txt
/usr/share
/usr/share/doc
/usr/share/doc/gvm-tools-1.4.1
/usr/share/doc/gvm-tools-1.4.1/README.rst.bz2
```

juippis commented on net-analyzer/gvm-tools/gvm-tools-1.4.1.ebuild:
```diff
@@ -0,0 +1,47 @@
+	sed -i "s/packages=find_packages(),.*/packages=find_packages(exclude=['tests*', 'docs']),/" "$S"/setup.py || die
+}
+
+src_configure() {
```
This ebuild should work even if you remove all these "empty" phases, configure, compile and install.

hsntgm replied:
I cleaned up empty phases.

juippis commented on net-analyzer/openvas-libraries/openvas-libraries-9.0.3.ebuild:
```diff
@@ -0,0 +1,91 @@
+	${DEPEND}"
+
+BDEPEND="
+	app-doc/doxygen[dot]
```
Depend conditionally with extras? USE flag?

hsntgm replied:
You are right.Fixed.

juippis commented on net-analyzer/openvas-manager/openvas-manager-7.0.3.ebuild:
```diff
@@ -0,0 +1,87 @@
+	${DEPEND}"
+
+BDEPEND="
+	app-doc/doxygen[dot]
```
Conditional depend?

hsntgm replied:
You are right.Fixed.

juippis commented on net-analyzer/openvas/files/README.gentoo:
```diff
@@ -1,34 +1,55 @@
-
-# Start the Greenbone Security Assistant
-rc-service gsad start or systemctl start gsad
+----------------------------------------------------------------------------------------------------------------------
```
Okay, this was pretty nice, but it wraps badly and becomes messy. You should use the default terminal width to wrap these lines, and maybe make it a bit more easily readable. I think the default was 72 or 80 characters, it's in the devmanual.

hsntgm replied:
fixed to 72 characters.

juippis commented:
All your 
```
einfo "It seems everything is going well."
einfo "Starting a full doc compile this will take some time."
```
are redundant in my opinion, since they show up for less than a second and get drowned in scrolling text immediately :P 

just a note, doesn't require action.

hsntgm commented:
> This is also BDEPEND.

You mean "sys-devel/gettext" is a build depend.It seems i am missing something here :)

```
net-analyzer/greenbone-security-assistant/greenbone-security-assistant-7.0.3.ebuild

BDEPEND="
	extras? ( app-doc/doxygen[dot] )
        sys-devel/gettext
	virtual/pkgconfig"
```

> EDIT: Also judging by the installed list files, this needs setuptools in RDEPEND.

A bit confused here.I suppose distutils-r1 has already setuptools in BDEPEND.Why this needs in RDEPEND juippis?

> You probably should remove the PDEPEND part from any other ebuild except openvas-libraries, since the other ebuilds depend on the libraries already that PDEPENDs on openvas.

openvas-scanner & openvas-manager & greenbone-security-agent depend to openvas-libraries.
ospd & gvm tools are not.

This is the issue that we have to decide which is best.The truth is avarage user definetly needs greenbone-security-agent(gsa).Because gsa is the WebUI for scanner.Expert users can manage things without gsa (WebUI).They can use command line interfaces (ospd & gvm-tools).

If i remove all PDEPEND expect openvas-libraries these are the example scenarios;

For example **if someone emerge openvas-manager directly** instead of main openvas package --> gsa(WebUI) & ospd(scanner-wrapper) & gvm-tools(cli) will not be installed and this setup is completly useless.

**if someone emerge openvas-scanner directly** instead of main openvas package the situation is same.

For proper openvas installation users need at least openvas-libraries + openvas-scanner + openvas-manager || WebUI(gsa) or cli(ospd & gvm-tools) 

How should we continue about that?

> Seems like "extras" for openvs-manager will require `dev-perl/CGI` in DEPEND.
> EDIT: (I did get the warning when building that it will be deprecated in the next version, though, so keep it in mind)

I don't know how i did miss this one.It seems somehow dev-perl/CGI has already installed to my system before.I will check my test env. again.It seems SQL Translator needs that and mainly it is part of openvas-libraries. 
dev-perl/CGI added to openvas-libraries under extras flag.

> All your
> 
> ```
> einfo "It seems everything is going well."
> einfo "Starting a full doc compile this will take some time."
> ```
> are redundant in my opinion, since they show up for less than a second and get drowned in scrolling text immediately :P
> 
> just a note, doesn't require action.

You are right.When i first started to maintain this package i used ninja and this einfo has a meaning there but after i discovered emake it seems it is meaningless they show up for less than a second. :)
I removed them.

juippis commented:
> > This is also BDEPEND.
> 
> You mean "sys-devel/gettext" is a build depend.It seems i am missing something here :)

Yes, move it from DEPEND to BDEPEND. The ebuild I see in Github has it in DEPEND. I probably should've checked your overlay you linked earlier, but I wgetted what you offered in this PR :P 

> 
> A bit confused here.I suppose distutils-r1 has already setuptools in BDEPEND.Why this needs in RDEPEND juippis?

It's needed in build-time and runtime. I'm pretty sure it's installed in every target machine anyway (BDEPEND takes care of host), but to make sure, you should include it in RDEPEND.

> For proper openvas installation users need at least openvas-libraries && openvas-scanner && openvas-manager && ( WebUI (greenbone-security-assistant) || cli (gvm-tools ) )
> 
> How should we continue about that?

Well, I personally don't have strong opinion about this. It's just weird when you think about dependency solving, but to me, looks like you've given it a lot of thought already ;)

> > There was a QA notice from either openvas or openvas-libraries about unused CMAKE variable, CMAKE_BUILD_RADIUS I believe. Could you investigate that?
> 
> I know that it comes from openvas-libraries.
> I think it is harmless.It warns about build completed without radius support which is supposed to be default.

Okay, glad you're aware.

hsntgm commented:
All done.

I believe that this package is totaly under my control now. But i need more complicated one to get more experience.I will try to pick one which suits best to my knowledge or maybe you can suggest/pick a new challange for me. 
It was such a great experience working with you @juippis .Thank you for your mentorship!
When you want the beer's on me..

juippis commented:
> I believe that this package is totaly under my control now. But i need more complicated one to get more experience.I will try to pick one which suits best to my knowledge or maybe you can suggest/pick a new challange for me.

You intend to become the proxy-maintainer of these packages, so that's how it's supposed to be. Don't try to "search" for stuff to do, contribute in areas that you personally use, and need.

> It was such a great experience working with you @juippis .Thank you for your mentorship!
> When you want the beer's on me..

Don't thank me yet ;) it still needs to get merged to the tree by a member of proxy-maint team, I can't do that. But in my eyes it's almost there definitely. Will run tests and report back today some time.

juippis commented on net-analyzer/gvm-tools/gvm-tools-1.4.1.ebuild:
```diff
@@ -0,0 +1,37 @@
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+DEPEND="
```
Are these really needed when building, or just when running (ie should all of the items in this list be moved to RDEPEND)

hsntgm replied:
Fixed gvm-tools.
also ospd is fixed.same situation.

I am not sure still we need **dev-python/setuptools** here ?
```
RDEPEND="
	dev-python/defusedxml[${PYTHON_USEDEP}]
        dev-python/lxml[${PYTHON_USEDEP}]
        dev-python/paramiko[${PYTHON_USEDEP}]
        dev-python/pythondialog:0[${PYTHON_USEDEP}]
        dev-python/setuptools[${PYTHON_USEDEP}]
        !net-analyzer/openvas-cli"

DEPEND="
        ${RDEPEND}"
```

juippis replied:
Yes, leave it for RDEPEND. distutils-r1 puts it in BDEPEND for you, but not for RDEPEND.
(I have a feeling it's needed in RDEPEND to check runtime dependencies/optionalities)

juippis commented on net-analyzer/openvas-scanner/openvas-scanner-5.1.3.ebuild:
```diff
@@ -0,0 +1,95 @@
+	!net-analyzer/openvas-tools"
+
+BDEPEND="
+	extras? ( app-doc/doxygen[dot] )
```
minor styling thing, but conditionals should come after unconditionals. I saw a 2nd ebuild with the same BDEPEND list, don't remember which one it was.

hsntgm replied:
thanks for the tip.fixed all conditionals | unconditionals sortings

juippis commented on net-analyzer/openvas/openvas-9.0.0.ebuild:
```diff
@@ -0,0 +1,42 @@
+		>=net-analyzer/openvas-manager-7.0.3[extras]
+		>=net-analyzer/greenbone-security-assistant-7.0.3[extras]
+		>=net-analyzer/ospd-1.3.2[extras]
+		app-doc/doxygen[dot]
```
Drop these below, and control the build-time dependencies from their own ebuilds. 'extras' in here forcing [extras] is good, though.

hsntgm replied:
You are right.
app-doc/doxygen already controlled from their own ebuilds.
Dropped.

juippis commented on net-analyzer/openvas/openvas-9.0.0.ebuild:
```diff
@@ -0,0 +1,42 @@
+		>=net-analyzer/openvas-scanner-5.1.3[extras]
+		>=net-analyzer/openvas-manager-7.0.3[extras]
+		>=net-analyzer/greenbone-security-assistant-7.0.3[extras]
+		>=net-analyzer/ospd-1.3.2[extras]
```
Make a new condition for extras? ( ospd? ...

same for greenbone-security-assistant

hsntgm replied:
This is not clear for me.
Can you explain a bit?

juippis replied:
Currently, ospd is pulled when 'ospd' USE flag is enabled. However, it will pull it also with 'extras'. So you'd need to move this line under 
```
ospd? ( 
	>=net-analyzer/ospd-1.3.2 
	extras? ( >=net-analyzer/ospd-1.3.2[extras] )
)
```

Similarily with greenbone-security-assistant. This is fine-tuning the logic :P

juippis commented on net-analyzer/ospd/ospd-1.3.2.ebuild:
```diff
@@ -0,0 +1,43 @@
+PDEPEND="
+	>=net-analyzer/openvas-9.0.0"
+
+python_prepare_all() {
```
this ebuild should work without this phase.

hsntgm replied:
done

juippis commented on net-analyzer/ospd/ospd-1.3.2.ebuild:
```diff
@@ -0,0 +1,43 @@
+	distutils-r1_python_compile
+}
+
+python_install_all() {
```
this ebuild should work without this phase.

hsntgm replied:
done

juippis commented on net-analyzer/ospd/ospd-1.3.2.ebuild:
```diff
@@ -0,0 +1,43 @@
+IUSE="+extras"
+
+DEPEND="
+	dev-python/paramiko[${PYTHON_USEDEP}]
```
Sort this list please :)

hsntgm replied:
sorted :)

juippis commented:
I am also not a fan of that README after openvas installation. It's just too long. I do understand it's importance, but maybe you need to strip it down a bit? Also it has commands included for packages you might not even install, so maybe those should be printed after the specific packages, not the meta package? Also the manually written 'optfeatures' take too much space and can be shortened and printed with 'optfeature' instead. 

Also the DEPEND-PDEPEND stuff still bothers me, but I know you've given it a lot of thought already. Still, I feel like the PDEPEND's on other than openvas-libraries is wrong, but then again, I don't have time or environment to test how changing them would work. Would it work if you change DEPEND from openvas to RDEPEND instead?

hsntgm commented:
I agree that README is too long.I tried to give lots of information to user because it is hard to find these commands all in one place.No problem, absolutly i will shorten it.

> Also the DEPEND-PDEPEND stuff still bothers me

> Would it work if you change DEPEND from openvas to RDEPEND instead?

Yes this is the old maintainer way.He didn't use any PDEPEND also.But it seems old maintainer's package stucked at test phase for a long time and there is no much user/feedback for this package so i am not sure it is correct.I also asked him about package's current progress but he was busy.

If you think this is more logical i can do it and remove all PDEPEND expect openvas-libraries.But i have to test it start from scratch.

The an other solution maybe merge it to tree and waiting some user feedback to see something goes wrong.So far in my tests everything went well.

```
RDEPEND="
	radius? ( >=net-analyzer/openvas-libraries-9.0.3[radius] )
	ldap? ( >=net-analyzer/openvas-libraries-9.0.3[ldap] )
	>=net-analyzer/openvas-libraries-9.0.3
	>=net-analyzer/openvas-scanner-5.1.3
	>=net-analyzer/openvas-manager-7.0.3
	gsa? ( >=net-analyzer/greenbone-security-assistant-7.0.3 )
	cli? ( >=net-analyzer/gvm-tools-1.4.1 )
	ospd? ( >=net-analyzer/ospd-1.3.2 )
	extras? (
		>=net-analyzer/openvas-libraries-9.0.3[extras]
		>=net-analyzer/openvas-scanner-5.1.3[extras]
		>=net-analyzer/openvas-manager-7.0.3[extras]
		>=net-analyzer/greenbone-security-assistant-7.0.3[extras]
		>=net-analyzer/ospd-1.3.2[extras]
		app-doc/doxygen[dot]
		app-doc/xmltoman
		app-text/htmldoc
		dev-texlive/texlive-latexextra
	)"
```

hsntgm commented on net-analyzer/gvm-tools/gvm-tools-1.4.1.ebuild:
```diff
@@ -0,0 +1,37 @@
+
+RDEPEND="
+	${DEPEND}
+	dev-python/setuptools
```
definetly i need python ebuild devmanual reading :)
done.

hsntgm commented on net-analyzer/greenbone-security-assistant/greenbone-security-assistant-7.0.3.ebuild:
```diff
@@ -0,0 +1,90 @@
+	dev-libs/libgcrypt:0=
+	dev-libs/libxml2:2
+	dev-libs/libxslt
+	extras? ( dev-python/polib )
```
done.

juippis commented:
> The an other solution maybe merge it to tree and waiting some user feedback to see something goes wrong.So far in my tests everything went well.
> 

I'd say go with this, you've given it more thought and testing time, so it should be trusted.

hsntgm commented:
> I'd say go with this, you've given it more thought and testing time, so it should be trusted.

Thank you for your trust but your opinion is important for me.

1-) I changed skeleton and removed all PDEPEND (including openvas-libraries) and at the same time i managed to apply my package resolution idea partially (but enough) with RDEPEND.

2-) Because of changed skeleton main openvas-9.0.0 ebuild's DEPENDS changed to RDEPEND and all BDEPEND's moved to their own ebuilds.Main openvas-9.0.0 ebuild is now completly clean and only resolves other packages.

3-) I completly removed readme.gentoo-r1 from openvas-9.0.0 ebuild (it was my fault,already badly used and caused warnings which you pointed) and used eutils (optfeature) and forwarded user to long README file in --> /etc/openvas/OPENVAS.gentoo instead of print it.

4-) All other sortings and minor things fixed.

I think this solution make happy both of us.

juippis commented:
I'm just gonna make a single comment instead of many :P

I saw this `dev-perl/CGI` in BDEPEND multiple times. Please be extra careful about BDEPEND, since it can introduce build-breakages. Is CGI only needed for the perl that gets executed in the build machine, or are those perl modules used when compiling the binary in host machine? To me, lack of better knowledge, it should be in DEPEND.

Your `optfeature "ldapsearch from OpenLDAP utilities — retrieves information from LDAP dictionaries" net-nds/openldap` has a typo and isn't showed due to that. `net-nds`. Also, it's too long, you should make sure these messages don't exceed the 80 character mark. The message for `net-analyzer/ike-scan` is also way too long. 

```
 *   net-analyzer/nikto for Nikto — a web server scanning and testing tool
 *   net-analyzer/nmap for NMAP — a portscanner
 *   net-analyzer/ike-scan for ike-scan - an IPsec VPN scanning, fingerprinting and testing tool
 *   net-analyzer/amap for amap — an application protocol detection tool
 *   app-forensics/ovaldi for ovaldi (OVAL) — an OVAL Interpreter
 *   net-analyzer/portbunny for portbunny — a Linux-kernel-based portscanner
 *   net-analyzer/w3af for w3af — a web application attack and audit framework
```

Could all be shortened to for example: `optfeature "web server scanning and testing tool" net-analyzer/nikto`
You get the idea, you dont have to repeat the optional PN here :P 

Then your openvas RDEPEND and even IUSE arent sorted alphabetically, 

```
radius? ( >=net-analyzer/openvas-libraries-9.0.3[radius] )
ldap? ( >=net-analyzer/openvas-libraries-9.0.3[ldap] )
```
should be below unconditionals, and then those conditionals sorted alphabetically. 
And you haven't updated the logic, which still pulls every dep with "extras" USE flag, even though they are controlled with "ospd" and "gsa" before. I can write it for you, if it doesn't make any sense, just let me know. 

`/etc/openvas/OPENVAS.gentoo` I don't think `/etc` is the correct place for this. Maybe put it under `/usr/share/openvas`?

Everything else looks good to me! And everything compiles, and seems to run just fine.

hsntgm commented:
I'm glad we're near to finish this.

I am fixing the issues you pointed.I understand your "tuning logic" part for openvas flags.I will apply it.
Sortings also will be done.

All 'dev-perl/CGI' will go to DEPEND.This one will deprecated next version also.

optfeature will be shortened.

I will put OPENVAS.gentoo readme to /usr/share/openvas.

hsntgm commented:
+ with yours, i discovered an other small issue about openvas flags and added fix to this pull request.There isn't any special changes other than we discussed. 

The problem was if users don't want to compile openvas with extras flag (recommended) other ebuilds still pull extras because they have default +extras flag which forces to use it.

I changed all "+extras" to "extras" in ebuilds (below) and now with your fine tuning logic everything works like a charm :)
```
openvas-libraries
openvas-scanner
openvas-manager
greenbone-security-assistant
ospd
```
After 1 month work for this package i feel in my bones that making package for gentoo is harder than setup lustre server or 100 node HPC :) But it was great experience for me.

If anyone wanna test this package from proxy-maint team (unfortunately I don't know anyone from gentoo team expect juippis) my overlay is always sync with latest.

https://github.com/hsntgm/openvas-9-gentoo-overlay

P.S.
juippis the beer's on me still :)

juippis commented:
> 
> After 1 month work for this package i feel in my bones that making package for gentoo is harder than setup lustre server or 100 node HPC :) But it was a great experience for me.

Yes I'm sorry for that also. Trying to help in making it perfect before anyone takes a look, so it could be merged right away. Also it is a huge PR to start with, with 50+ files modified. Easier to start with something smaller ;)

> 
> If anyone wanna test this package from proxy-maint team (unfortunately I don't know anyone from gentoo team expect juippis) my overlay is always sync with latest.

Well I'm not actually part of any Gentoo team, just a little helper trying to make things easier and faster for devs, and trying to get user contributions up to quality and into the tree!

Looks good to me! Now we need someone from the proxy-maint team to take a look. But I don't see any more mistakes in this that we could deal with. It is an intimidating PR for anyone to take a look at, and I feel like the devs generally avoid "new package" tags. This however has bumps for 6 currently in-tree packages, too. 

Our time together has come to an end. I hope some member of proxy-maint will take a look, I can try to ping people for that, but I can't guarantee it. Good luck! 

> 
> P.S.
> juippis the beer's on me still :)

I'll gladly take on that offer ;)
